docs/V1/TokenizationAndCompilation.md

```diff
@@ -0,0 +1,96 @@
+  e.g. `SpecialFactory` 
+* {Actor Short Pascal Case Name}:
+  Represents a call to `$this->getActor()->getShortPascalCaseName()`
+  e.g. `Factory`
+
+|Type|Regex/String|Replacement|
+|---|---|---|
+|Regex|`namespace(\s+)Neighborhoods\\BuphaloTemplateTree\\PrimaryActorName`|`namespace **NAMESPACE_PREFIX_TOKEN****NAMESPACE_RELATIVE_TOKEN**\**PRIMARY_ACTOR_SHORT_PASCAL_CASE_NAME_TOKEN**`|
+|Regex|`namespace(\s+)Neighborhoods\\BuphaloTemplateTree`|`namespace **NAMESPACE_PREFIX_TOKEN****NAMESPACE_RELATIVE_TOKEN**`|
+|Regex|`use(\s+)Neighborhoods\\BuphaloTemplateTree\\PrimaryActorName`|`use **NAMESPACE_PREFIX_TOKEN****NAMESPACE_RELATIVE_TOKEN**\**PRIMARY_ACTOR_SHORT_PASCAL_CASE_NAME_TOKEN**`|
+|Regex|`use(\s+)Neighborhoods\\BuphaloTemplateTree/`|`use **NAMESPACE_PREFIX_TOKEN**`|
+|Regex|`services:\n(\s+)Neighborhoods\\BuphaloTemplateTree\\PrimaryActorName`|`services:\n  **NAMESPACE_PREFIX_TOKEN****NAMESPACE_RELATIVE_TOKEN**\**PRIMARY_ACTOR_SHORT_PASCAL_CASE_NAME_TOKEN**`|
+|Regex|`class:(\s+)Neighborhoods\\BuphaloTemplateTree\\PrimaryActorName`|`class: **NAMESPACE_PREFIX_TOKEN****NAMESPACE_RELATIVE_TOKEN**\**PRIMARY_ACTOR_SHORT_PASCAL_CASE_NAME_TOKEN**`|
+|String|`['@Neighborhoods\BuphaloTemplateTree\PrimaryActorName`|`['@**NAMESPACE_PREFIX_TOKEN****NAMESPACE_RELATIVE_TOKEN**\**PRIMARY_ACTOR_SHORT_PASCAL_CASE_NAME_TOKEN**`|
+|String|`\PrimaryActorName;`|`**EMPTY_TOKEN**;`|
+|String|`\{Template File Name};`|`**EMPTY_TOKEN**;`|
+|String|`\PrimaryActorNameInterface`|`\**PRIMARY_ACTOR_SHORT_PASCAL_CASE_NAME_TOKEN**Interface`|
+|String|`\{Template File Name}Interface`|`\**SHORT_PASCAL_CASE_NAME_TOKEN**Interface`|
+|String|`\PrimaryActorName\`|`**EMPTY_TOKEN**\`|
+|String|`\PrimaryActorName`|`\**RELATIVE_CLASS_PATH_TOKEN**`|
+|String|`\{Template File Name}`|`\**SHORT_PASCAL_CASE_NAME_TOKEN**`|
+|Regex|`protected(\s+)\$PrimaryActorName`|`protected $**PRIMARY_ACTOR_FULL_PASCAL_CASE_NAME_TOKEN**`|
+|Regex|`protected(\s+)\${Template File Name}`|`protected $**PRIMARY_ACTOR_FULL_PASCAL_CASE_NAME_TOKEN**`|
+|String|`$this->PrimaryActorName`|`$this->**PRIMARY_ACTOR_FULL_PASCAL_CASE_NAME_TOKEN**`|
+|String|`$this->{Template File Name}`|`$this->**PRIMARY_ACTOR_FULL_PASCAL_CASE_NAME_TOKEN**`|
+|Regex|`class(\s+)PrimaryActorName`|`class **PRIMARY_ACTOR_SHORT_PASCAL_CASE_NAME_TOKEN**`|
+|Regex|`class(\s+){Template File Name}`|`class **SHORT_PASCAL_CASE_NAME_TOKEN**`|
+|String|`$PrimaryActorName`|`$**PRIMARY_ACTOR_SHORT_PASCAL_CASE_NAME_TOKEN**`|
+|String|`${Template File Name}`|`$**PRIMARY_ACTOR_SHORT_PASCAL_CASE_NAME_TOKEN**`|
+|Regex|`interface(\s+)PrimaryActorNameInterface`|`interface **PRIMARY_ACTOR_SHORT_PASCAL_CASE_NAME_TOKEN**Interface`|
+|Regex|`interface(\s+){Template File Name}`|`interface **SHORT_PASCAL_CASE_NAME_TOKEN**`|
+|String|`PrimaryActorNameInterface`|`**PRIMARY_ACTOR_SHORT_PASCAL_CASE_NAME_TOKEN**Interface`|
+|String|`{Template File Name}Interface`|`**SHORT_PASCAL_CASE_NAME_TOKEN**Interface`|
```

Comment:
It would seem like these can be "combined" with the versions that have preceding `\`s (or really those earlier rows can be omitted?)
but if there's an implied `^` or `\s` in front of each of these, then nevermind
or there may be some other reason these should be separate, e.g. this is a literal catalog of what the code does and the code treats these separately, despite the overlap. thus a reason for a v2.

Reply:
> this is a literal catalog of what the code does and the code treats these separately, despite the overlap. thus a reason for a v2.

☝🏻 

Yup. The current code is a mess and a lot of it could _probably_ be simplified in ways that would not result in any functional changes.